packet.go

```diff
@@ -256,8 +327,15 @@ func (p *Packet) decodeUdp() {
 	udp.Length = binary.BigEndian.Uint16(pkt[4:6])
 	udp.Checksum = binary.BigEndian.Uint16(pkt[6:8])
 	p.setHeader(udp)
-
 	p.Payload = pkt[8:]
+
+	// potential vxlan packet. vxlan header is 8 bytes.
+	// typical ports are IANA assigned 4789 or linux default 8472.
```

Comment:
You could add a filter check before specifically comparing the values in an array. 
For example doing a (given udp.DestPort  & 0b10000 )  might be cheaper than doing the scan, as there is one common bit between both and that has to be true if the incoming port has to match either of them.

python3
>>> "{0:b}".format(4789)
'1001010110101'
>>> "{0:b}".format(8472)
'10000100011000'
>>> "{0:b}".format(4789&8472)
'10000'

(2) just doing port == val1 || port == val2 might be cheaper than array based scan.

This is per packer optimization and can wait I think.

Reply:
vxlan_ports are now passed by caller and could be completely different from 4789 or 8472. Thanks for suggesting it and the code snippet. I'm going to steal it.